starboard/raspi/shared/launcher.py

```diff
@@ -188,19 +193,38 @@ def _PexpectSpawnAndConnect(self, command):
           self._PexpectSendLine('echo ' + Launcher._SSH_LOGIN_SIGNAL)
           i = self.pexpect_process.expect([Launcher._SSH_LOGIN_SIGNAL])
           break
-      except pexpect.TIMEOUT:
+      except (pexpect.TIMEOUT, OSError, pexpect.ExceptionPexpect) as err:
         if self.shutdown_initiated.is_set():
           return
         retry_count += 1
         # Check if the max retry count has been exceeded. If it has, then
         # re-raise the timeout exception.
         if retry_count > Launcher._PEXPECT_PASSWORD_TIMEOUT_MAX_RETRIES:
-          raise
+          raise RuntimeError(f'Failed after {retry_count} retries') from err
+
+  def _PexpectSpawnAndConnect(self, command):
+    """Outer retry wrapper"""
+    retry_count = 0
```

Comment:
Won't this ever only be 0 or 1?

Reply:
If i don't add the loops, then yeah, which i didn't intend. Thanks for calling out